tests/System.Text.Utf8String.Tests/RandomTests.cs

```diff
@@ -631,6 +631,37 @@ public void IndexOfTests(int expected, string s, string substring)
         }
 
         [Theory]
+        [InlineData(0, "", "")]
+        [InlineData(0, "abc", "")]
+        [InlineData(-1, "", "a")]
+        [InlineData(-1, "", "abc")]
+        [InlineData(0, "a", "a")]
+        [InlineData(-1, "a", "b")]
+        [InlineData(0, "abc", "a")]
+        [InlineData(0, "abc", "ab")]
+        [InlineData(0, "abc", "abc")]
+        [InlineData(1, "abc", "b")]
+        [InlineData(1, "abc", "bc")]
+        [InlineData(3, "abcc", "c")]
+        [InlineData(3, "abcbc", "bc")]
+        [InlineData(38, "aaaaaaabaaaaaaaabaaaaaaaabaaaaaaaabaaaaaaaaaaab", "aaaaaaaab")]
+        [InlineData(16, "abbbbbbbabbbbbbbabbbbbbbbbbbbbbbbbbbbbbbbbbbb", "abbbbbbbb")]
+        [InlineData(-1, "aaabaaa", "aabaaaa")]
+        [InlineData(-1, "aaabaaa", "aadaaaa")]
+        [InlineData(3, "ababababa", "bababa")]
+        [InlineData(-1, "abababab", "bbababa")]
+        [InlineData(-1, "abababab", "bababaa")]
+        [InlineData(7, "aaabaacaaac", "aaac")]
+        [InlineData(9, "baaaabaaabaaaa", "baaaa")]
+        [InlineData(0, "\uABC0\uABC1\uABC2", "\uABC0\uABC1")]
```

Comment:
Could you add a case that uses codeponts from outside of the BMP, but one that does not return 0? Thanks!